src/persistence/SubjectOperationExecutor.ts

```diff
@@ -89,8 +89,8 @@ export class SubjectOperationExecutor {
             !this.updateSubjects.length &&
             !this.removeSubjects.length &&
             !this.relationUpdateSubjects.length &&
-            !subjects.some(subject => !subject.junctionInserts.length) &&
-            !subjects.some(subject => !subject.junctionRemoves.length))
+            !subjects.some(subject => !!subject.junctionInserts.length) &&
+            !subjects.some(subject => !!subject.junctionRemoves.length))
```

Comment:
also can you please add tests for this issue. I think you need to check cases when there are junction inserts/removes and there no, check each of them separately and make sure that "return" is executed when there are no operations (and transaction isnt started, thats why this check is here). To simplify and isolate testing you can extract this method into separate class I think.